substrate/frame/support/procedural/src/derive_impl.rs

```diff
@@ -56,18 +56,59 @@ fn is_runtime_type(item: &syn::ImplItemType) -> bool {
 		false
 	})
 }
-
-#[derive(Parse, Debug)]
 pub struct DeriveImplAttrArgs {
 	pub default_impl_path: Path,
+	pub generics: Option<AngleBracketedGenericArguments>,
 	_as: Option<Token![as]>,
-	#[parse_if(_as.is_some())]
 	pub disambiguation_path: Option<Path>,
 	_comma: Option<Token![,]>,
-	#[parse_if(_comma.is_some())]
 	pub no_aggregated_types: Option<keyword::no_aggregated_types>,
 }
 
+impl syn::parse::Parse for DeriveImplAttrArgs {
+	fn parse(input: syn::parse::ParseStream) -> Result<Self> {
+		let mut default_impl_path: Path = input.parse()?;
+		// Extract the generics if any
+		let (default_impl_path, generics) = match default_impl_path.clone().segments.last() {
+			Some(PathSegment { ident, arguments: PathArguments::AngleBracketed(args) }) => {
```

Comment:
Any other arguments beside `None` should result in an error?

Reply:
> Any other arguments beside `None` should result in an error?

Makes sense. Let me add that in.